tests/static_analyzer/test_reference_resolve_real_refs.py

```
@@ -0,0 +1,465 @@
+from static_analyzer.analysis_result import StaticAnalysisResults
+from static_analyzer.node import Node
+from static_analyzer.reference_resolve_mixin import ReferenceResolverMixin
+
+
+class ConcreteResolver(ReferenceResolverMixin):
+    """Concrete test implementation of the mixin."""
+
+    def __init__(self, repo_dir: Path, static_analysis: StaticAnalysisResults):
+        super().__init__(repo_dir, static_analysis)
+
+    def _parse_invoke(self, prompt, type):
```

Ruff (0.15.5)

[error] Function argument `type` is shadowing a Python builtin

(A002)